src/components/ScreenWrapper.js

```diff
@@ -35,6 +39,12 @@ const propTypes = {
     // Called when navigated Screen's transition is finished.
     onTransitionEnd: PropTypes.func,
 
+    /** Is the window width narrow, like on a mobile device */
+    isSmallScreenWidth: PropTypes.bool.isRequired,
+
+    /** The styles to pass to the KeyboardAvoidingView i.e. height/padding */
+    keyboardAvoidingViewBehavior: PropTypes.string,
```

Comment:
The [official RN documentation](https://reactnative.dev/docs/keyboardavoidingview#behavior) doesn't really give any insight. The most helpful thing I could find directly from RN was [this switch statement on `behavior`](https://github.com/facebook/react-native/blob/92073d4a71d50a1ed80cf9cb063a6144fcc8cf19/Libraries/Components/Keyboard/KeyboardAvoidingView.js#L157), which tells us how it interprets the `KeyboardAvoidingView` depending on which behavior is specified.

I read about this a good deal online, most sources say that you should use `'padding'` for iOS, but they don't mention much about Android. Most people are saying that it is layout/device dependent, and that you should test each of them on each device to ensure they aren't causing any problems. I'll leave a link to the

Reply:
Got it. Any particular reason why we are defaulting to `'padding'` and then sometimes switching it to `'height'` in this PR? I guess my ultimate concern is that if there is some preference to use one or the other I would like to know whether we should be thinking about it or if it is safe to default to `'padding'` in most cases. This isn't a blocker though - just trying to educate myself. Thanks!

Reply:
takeaways:

- padding

>If you don’t have a lot of information on your screen then padding is likely to be the best choice because the whole screen is available to view, regardless of whether every view is resized or not.

- position

>When using the position option, the entire view will be moved ‘up’

Unsure how this is different from adding the padding, but looks like it moves up things even if they are not wrapped in the `KeyboardAvoidingView`

- `height`

>height behaviour will resize the view on the screen after the keyboard pop up. To use height, the component should be placed at the start of the high order view after the resize of ‘KeyboardAvoidingView’. You’ll also need a little help from the ‘KeyboardVerticalOffset’ props

didn't quite understand this paragraph but seems like if we pass `height` we should also pass a "vertical offset"

The author does say:

>our first choice is always padding (particularly if you don’t have many components in the screen, so remember to always test with smaller screens)

So I feel good about that as the default, but still not feeling 100% confident haha.

Reply:
The main reason I was using `padding` as the default was because we have [this component](https://github.com/Expensify/App/blob/main/src/components/KeyboardSpacer/index.ios.js) which defaults to padding which we were basically using everywhere. This component basically only enables the `KeyboardAvoidingView` on iOS and disables it everywhere else, and passed `behavior="padding"`.